pkg/nodegraph/Library/NodegraphRecipe.js

```diff
@@ -98,6 +108,18 @@ export const NodegraphRecipe = {
     },
     newNodeInfos: {
       $type: '[Pojo]'
+    },
```

Comment:
Suggest factoring this stuff (in one of a few ways). There are Stores and Particles that go together, whenever possible if we can extract them into little recipes that we can add to the main recipe, this is better than ever-growing the main recipe.

Ideally we make mini-recipes and just slot-them in. If that's not convenient, we can at least create mixin objects. 

I think this process is something we can think about and improve.

Reply:
totally agree!
PS: just some general thoughts - the smaller the particle, the more it interacts with other particles, the more connections it needs. Sometimes it feels like too many connections (like when a function takes too many parameters), I don't know if this is a concern, or if there is a healthy balance.

Also for example, Editor doesn't need read access to the `events` store, it only appends to the list.. maybe we could improve granularity of access permissions (in the future)